### PR TITLE
fix(content-manager): translate content-type fields in List and Edit …

### PR DESCRIPTION
**Pull request by markkaylor**
### What does it do?

- Adds ability to translate a content-types' attributes in both the list view (table headers) and edit view (form labels)
- Adds ability to translate components' attributes wherever the component is used

The translation ids are computed with:

`content-manager.{content-types | components}.{contentTypeUid | componentUid}.{attributeName}`

### Why is it needed?

- It was possible in Strapi 4

### How to test it?

In `examples/getstarted/src/admin/app.jsx` provide a config like the one below.
Then change the language of your app in your profile to one of the available locales

```js
const config = {
  locales: ['it', 'es', 'en', 'en-GB', 'fr'],
  translations: {
    fr: {
      // Keys scoped to a content-type, tranlsates the attribute for that content-type
      'content-manager.content-types.api::kitchensink.kitchensink.short_text': 'translated short text',
      'content-manager.content-types.api::kitchensink.kitchensink.single_compo':
        'translated single compo',
      // Keys scoped to a component, translates the component attribute for any content-type that uses it
      'content-manager.components.basic.relation.categories': 'translated dynamic zone compo attribute',
      'content-manager.components.default.temp.name': 'translated default.temp name',
      'content-manager.components.basic.simple.name': 'translated basic.simple name',
      'content-manager.components.basic.simple.nested': 'translated nested compo',
    },
  },
};
```
- Providing a translation for a root level attribute (text, component, dynamic zone, integer, etc...) should only translate within the scope of its content-type
- Providing a translation for a component (non-repeatable, repeatable, and dynamic zone) should translate within the scope of the component and therefore in _any_ content-type that is using it.

### Related issue(s)/PR(s)

resolves: https://github.com/strapi/strapi/issues/20461

### CODE REVIEW

File: packages/core/content-manager/admin/src/pages/EditView/components/FormLayout.tsx [hunk 1]
```diff
@@ -3,20 +3,36 @@ import { Box, Flex, Grid } from '@strapi/design-system';
...
+import { useIntl } from 'react-intl';
 
 interface FormLayoutProps extends Pick<EditLayout, 'layout'> {}
 
 const FormLayout = ({ layout }: FormLayoutProps) => {
+  const { formatMessage } = useIntl();
+
+  const getTranslatedLabel = (label: string) => {
+    return formatMessage({
+      id: `content-manager.containers.EditPage.form.${label}`,
```
Comment (markkaylor) [inline packages/core/content-manager/admin/src/pages/EditView/components/FormLayout.tsx:15]: ❓  I'm wondering if we should scope the translations to a content-type and not worry about the page we are on? `content-manager.container.{content-type-name}.{attribute-name}`.

Comment (remidej) [reply packages/core/content-manager/admin/src/pages/EditView/components/FormLayout.tsx:15]: That sounds nice, but I the same time I think it means breaking backwards compatibility? If so I think we can keep the current ids

**Review by remidej — APPROVED** [review]
haven't done QA but LGTM :)

File: packages/core/content-manager/admin/src/pages/EditView/components/FormLayout.tsx (hunk 1 above)
Comment (remidej) [reply packages/core/content-manager/admin/src/pages/EditView/components/FormLayout.tsx:15]: if we need a BC anyway then let's make the best api then 🚀 I agree the label should be the same between pages

**Comment by markkaylor** [comment]
@remidej I pushed to use a single key for both pages: `content-manager.{contentTypeUid}.{attributeName}`

```
const config = {
  locales: ['it', 'es', 'en', 'en-GB', 'fr'],
  translations: {
    fr: {
      'content-manager.api::kitchensink.kitchensink.short_text': 'texte_court'
    }
  }
};
```

File: packages/core/content-manager/admin/src/pages/EditView/components/FormLayout.tsx [hunk 2]
```diff
@@ -1,22 +1,40 @@
...
 
 interface FormLayoutProps extends Pick<EditLayout, 'layout'> {}
 
 const FormLayout = ({ layout }: FormLayoutProps) => {
+  const { formatMessage } = useIntl();
+  const { model } = useDoc();
+
+  const getTranslatedLabel = (label: string) => {
+    return formatMessage({
+      id: `content-manager.${model}.${label}`,
```
Comment (remidej) [inline packages/core/content-manager/admin/src/pages/EditView/components/FormLayout.tsx:17]: This is much better but I think we could still have collisions 🤔 how about `content-manager.[something].${model}.${label}`, with something like field in the middle?

Comment (markkaylor) [reply packages/core/content-manager/admin/src/pages/EditView/components/FormLayout.tsx]: `content-manager.content-types.${model}.${label}`? wdyt? All ears for another suggestion, before it was containers but I don't like that

Comment (remidej) [reply packages/core/content-manager/admin/src/pages/EditView/components/FormLayout.tsx]: sounds good!

**Comment by markkaylor** [comment]
@remidej updated

**Review by remidej — APPROVED** [review]
Works nicely! Just a heads up you're missing `.content-types` in the PR description key (I think Marion uses them for QA around release time, that's why it's good to keep it updated)

Also I wonder if this should handle components too? If so, how? And should it be another PR?

**Comment by markkaylor** [comment]
> Works nicely! Just a heads up you're missing `.content-types` in the PR description key (I think Marion uses them for QA around release time, that's why it's good to keep it updated)
> 


My bad thanks for the heads up. I updated.

> Also I wonder if this should handle components too? If so, how? And should it be another PR?

I'll take a look at components now to see if it makes sense to handle it here.

**Comment by kauedelmonte** [comment]
Please release this 🙏

**Comment by markkaylor** [comment]
@remidej I don't think there is anything to be done here for components actually. Maybe something in the CTB? but it looks like there isn't any translation over there and the attribute name is used directly. I think I'll go ahead and merge this.

File: packages/core/content-manager/admin/src/pages/EditView/components/FormLayout.tsx [hunk 3]
```diff
@@ -1,22 +1,40 @@
...
   return (
     <Flex direction="column" alignItems="stretch" gap={6}>
       {layout.map((panel, index) => {
         if (panel.some((row) => row.some((field) => field.type === 'dynamiczone'))) {
           const [row] = panel;
           const [field] = row;
+
+          const fieldWithTranslatedLabel = {
+            ...field,
+            label: getTranslatedLabel(field.label),
```
Comment (markkaylor) [inline packages/core/content-manager/admin/src/pages/EditView/components/FormLayout.tsx:31]: This is wrong I should be passing field.name

File: packages/core/content-manager/admin/src/pages/EditView/components/FormLayout.tsx [hunk 4]
```diff
@@ -1,22 +1,40 @@
...
 
 import { InputRenderer } from './InputRenderer';
 
 interface FormLayoutProps extends Pick<EditLayout, 'layout'> {}
 
 const FormLayout = ({ layout }: FormLayoutProps) => {
+  const { formatMessage } = useIntl();
+  const { model } = useDoc();
+
+  const getTranslatedLabel = (label: string) => {
```
Comment (markkaylor) [inline packages/core/content-manager/admin/src/pages/EditView/components/FormLayout.tsx:15]: the argument should be name, not label

**Review by remidej — APPROVED** [review]
Tested and works well!

I'm noticing the labels don't apply to history, but I think that makes sense, since we don't track the history of those labels over time

**Pull request merged**